CMake/portability/CheckCXXStandardSupport.cmake

```diff
@@ -22,3 +22,14 @@ endif()
 check_cxx_compiler_flag("${CMAKE_CXX11_STANDARD_COMPILE_OPTION}" COMPILER_SUPPORTS_CXX11)
 check_cxx_compiler_flag("${CMAKE_CXX14_STANDARD_COMPILE_OPTION}" COMPILER_SUPPORTS_CXX14)
 check_cxx_compiler_flag("${CMAKE_CXX17_STANDARD_COMPILE_OPTION}" COMPILER_SUPPORTS_CXX17)
+
+function(select_recent_cxx_standard_available)
```

Comment:
IMHO I find too intrusive to modify the parent environment. The `message(FATAL_ERROR ...)` in top-level CMakeLists.txt is enough.

Reply:
It only sets when user requests it with `select_recent_cxx_standard_available` and only if CMAKE_CXX_STANDARD is not set yet. Parent scope is the caller scope so the user gets what he asked, no side-effects AFAIK. Do you see other problems? suggestions?

Reply:
I am fine with that then.